### PR TITLE
24 updating the readme file

### LINKED ISSUE #24: updating the README file
Populating the Readme file based on the guidelines.

### PR DESCRIPTION
**Pull request by atabak-alishiri**
The first draft is ready, please review and complete parts indicated by (TO BE ADDED)

### CODE REVIEW

**Comment by salva-u** [comment]
Suggestion to change from dataset to about general section:
"The objective of our analysis was to predict the learning ability of induviduals by way of predicting their scores on the english speaking test in the quiz ( to measure english proficiency) based on their demographic information and their languages background in terms of how much time they've spent in english speaking countries, their primary and native language, etc.
We attempt to build a regression model by using Ridge Model and Lasso Model and measure the model's performance by using R-squared, Root Mean Squared Error, Negative Mean Squared Error. ADD DETAILS ABOUT MODEL CONCLUSIONS...

The dataset we used was collected from 680,333 ( between the ages of 7 and 89) participants were asked to fill an english grammar quiz, additionally their demographic information, and additional details about their language background and countries they've lived in was collected. This dataset was used in the research paper titled "A critical period for second language acquisition: Evidence from 2/3 million
English speakers" by Joshua K. Hartshornea,b,⁎, Joshua B. Tenenbauma, Steven Pinkerc. who collected the dataset described above in order to study the ultimate attainment of grammar-learning ability, specifically looking at the preservation of this ability throughout childhood and its rapid decline in late adolescence. The authors conducted analyses and compared various models to determine the age-related patterns of language acquisition.The dataset can be found here https://osf.io/pyb8s/. We had to use a subset ( 30% of full dataset) of the dataset for milestone 1 the script used to obtain the subset can be found in `src/Sampling from dataset.ipynb`, during later releases of the project we would consider using the full dataset."

**Comment by salva-u** [comment]
can we add another dependency same as the milestone draft example:
Python and packages listed in [environment.yml]

**Comment by atabak-alishiri** [comment]
@farrandi Can you please complete the readme file based on the results?

**Comment by rbouwer** [comment]
Looks good! Will review again once finalized

**Comment by farrandi** [comment]
Added the results and discussion section (similar to PR #32 ). Here is the text I added:

Our analysis, based on a dataset of approximately 200,000 individuals, shows that factors such as age, education, and language background significantly predict English proficiency. The regression model achieved a 5.3% RMSE on test data, confirming the reliability of these predictors in assessing language skills. Notably, being a native English speaker emerged as the strongest positive predictor, while immersion in English learning showed a strong negative correlation. These findings reinforce that demographic and educational backgrounds are crucial in language acquisition. This opens avenues for future research, particularly in understanding how cultural exposure influences language proficiency, potentially leading to more effective and personalized language learning strategies.

**Review by salva-u — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,12 +1,91 @@
...
+
+---
+
+## 🔍 Methodology
+
+We employ a comprehensive approach:
+
+1. **Data Preprocessing:** Cleaning, normalization, and transformation of data to ensure quality and consistency.
+2. **Exploratory Data Analysis:** Utilizing statistical techniques and visualization tools to uncover trends and patterns.
+3. **Model Development:** TO BE ADDED
```
Comment (rbouwer) [inline README.md]: This still says TO BE ADDED

**Review by rbouwer — COMMENTED** [review]
Left comment about TO BE ADDED remaining in README

File: README.md (hunk 1 above)
Comment (atabak-alishiri) [reply README.md]: Thank you for mentioning that, I updated that one

File: README.md [hunk 2]
```diff
@@ -1,12 +1,91 @@
...
+---
+
+## 🔍 Methodology
+
+We employ a comprehensive approach:
+
+1. **Data Preprocessing:** Cleaning, normalization, and transformation of data to ensure quality and consistency.
+2. **Exploratory Data Analysis:** Utilizing statistical techniques and visualization tools to uncover trends and patterns.
+3. **Model Development:** We developed predictive models by employing Ridge and Lasso regression techniques to ascertain the weight of each variable in predicting English language proficiency. By fine-tuning these models, we aimed to minimize prediction errors and enhance the accuracy of our forecasts.
+4. **Model Evaluation:** Our models were rigorously evaluated using cross-validation techniques and a variety of performance metrics such as R-squared, Root Mean Squared Error (RMSE), and Negative Mean Squared Error (NMSE), ensuring their robustness and reliability in predicting language learning ability.
```
Comment (rbouwer) [inline README.md]: Take out NMSE part - we didn't use this

Comment (atabak-alishiri) [reply README.md]: Done

**Review by rbouwer — APPROVED** [review]
Great! Thanks for making those changes :)

**Review by farrandi — APPROVED** [review]

**Pull request merged**